### PR TITLE
update torch seed

### PR DESCRIPTION
**Pull request by johnnv1**
Related to #2027 -- I think this patch doesn't cover all the failed tests, but that's what I can find in the logs of the last PR (if failures continue to occur related to this case, we should try to note in the issue #2027)

The issue is with `_torch_solve_cast` which will fail [if the A matrix is not invertible or any matrix in a batched A is not invertible.](https://pytorch.org/docs/stable/generated/torch.linalg.solve.html)

### CODE REVIEW

File: test/tracking/test_planar_tracking.py [hunk 1]
```diff
@@ -38,11 +38,15 @@ def test_real(self, device, dtype, data):
             if isinstance(data[k], torch.Tensor):
                 data[k] = data[k].to(device, dtype)
         h0, w0 = data["image0"].shape[2:]
-        data["image0"] = resize(data["image0"], (int(h0 // 2), int(w0 // 2)))
-        data["image1"] = resize(data["image1"], (int(h0 // 2), int(w0 // 2)))
+        data["image0"] = resize(
```
Comment (edgarriba) [inline test/tracking/test_planar_tracking.py:41]: you ca use rescale with factor 0.5

File: test/test_contrib.py [hunk 2]
```diff
@@ -415,6 +415,7 @@ def test_gradcheck(self, device, dtype):
 class TestImageStitcher:
     @pytest.mark.parametrize("estimator", ['ransac', 'vanilla'])
     def test_smoke(self, estimator, device, dtype):
+        torch.manual_seed(7)  # issue kornia#2027
```
[the remark this answers is not in the document]
Comment (johnnv1) [reply test/test_contrib.py:418]: Yep, just try here some seed which doesn't generate a case which fail at `torch.linalg.solve` -- some seeds generate cases which will fail to just one dtype, other for both

File: test/tracking/test_planar_tracking.py [hunk 3]
```diff
@@ -38,11 +38,15 @@ def test_real(self, device, dtype, data):
...
+            data["image0"], (int(h0 // 2), int(w0 // 2)), interpolation='bilinear', align_corners=False
+        )
+        data["image1"] = resize(
+            data["image1"], (int(h0 // 2), int(w0 // 2)), interpolation='bilinear', align_corners=False
+        )
         with torch.no_grad():
             tracker.set_target(data["image0"])
-            torch.random.manual_seed(0)
+            torch.manual_seed(3)  # issue kornia#2027
```
Comment (johnnv1) [inline test/tracking/test_planar_tracking.py:49]: maybe, here there must be a xfail -- the crash occurs related to ransac, I don't know if the seed will be enough to avoid crashes

**Comment by johnnv1** [comment]
maybe merge #2038, to test this also on mac

**Comment by johnnv1** [comment]
The macos failure cases are related to `ransac` (in `find_homography_dlt` - https://github.com/kornia/kornia/blob/667cf45b1cc9576018cdc070bf553b7fcd662e74/kornia/geometry/homography.py#L179  which calls https://github.com/kornia/kornia/blob/667cf45b1cc9576018cdc070bf553b7fcd662e74/kornia/utils/helpers.py#L178) -- same issue also occurs in `TestImageStitcher` and `TestHomographyTracker`

Maybe the better case here is to use xfail without seed?

**Comment by edgarriba** [comment]
i suggest adding hardcoded cases in the tests @ducha-aiki

**Comment by johnnv1** [comment]
I prefer too, I will check if the cases can be with small inputs

File: test/feature/test_integrated.py [hunk 4]
```diff
@@ -294,14 +294,14 @@ def test_real_sift_preextract(self, device, dtype, data):
     @pytest.mark.skipif(sys.platform == "win32", reason="this test takes so much memory in the CI with Windows")
     @pytest.mark.parametrize("data", ["loftr_homo"], indirect=True)
     def test_real_gftt(self, device, dtype, data):
-        torch.random.manual_seed(0)
         # This is not unit test, but that is quite good integration test
         matcher = LocalFeatureMatcher(GFTTAffNetHardNet(1000), DescriptorMatcher('snn', 0.8)).to(device, dtype)
         ransac = RANSAC('homography', 1.0, 1024, 5).to(device, dtype)
         data_dev = utils.dict_to(data, device, dtype)
         pts_src = data_dev['pts0']
```
Comment (johnnv1) [inline test/feature/test_integrated.py:301]: Already have a hardcoded input and is failing before

**Comment by johnnv1** [comment]
Tests are failing because #2068 -- I will remove the `fail-fast` on CI here to be able to check the failing tests we are targeting here

File: test/test_contrib.py [hunk 5]
```diff
@@ -415,9 +415,33 @@ def test_gradcheck(self, device, dtype):
 class TestImageStitcher:
     @pytest.mark.parametrize("estimator", ['ransac', 'vanilla'])
     def test_smoke(self, estimator, device, dtype):
-        B, C, H, W = 1, 3, 224, 224
-        sample1 = torch.rand(B, C, H, W, device=device, dtype=dtype)
-        sample2 = torch.rand(B, C, H, W, device=device, dtype=dtype)
+        B, C, H, W = 1, 3, 6, 6
```
Comment (johnnv1) [inline test/test_contrib.py:418]: Continue to fail on macos https://github.com/kornia/kornia/actions/runs/3639450296/jobs/6142845358#step:4:9575

File: test/tracking/test_planar_tracking.py [hunk 6]
```diff
@@ -38,11 +38,11 @@ def test_real(self, device, dtype, data):
...
                 data[k] = data[k].to(device, dtype)
         h0, w0 = data["image0"].shape[2:]
-        data["image0"] = resize(data["image0"], (int(h0 // 2), int(w0 // 2)))
-        data["image1"] = resize(data["image1"], (int(h0 // 2), int(w0 // 2)))
+        data["image0"] = rescale(data["image0"], 0.5, interpolation='bilinear', align_corners=False)
+        data["image1"] = rescale(data["image1"], 0.5, interpolation='bilinear', align_corners=False)
         with torch.no_grad():
             tracker.set_target(data["image0"])
-            torch.random.manual_seed(0)
+            torch.manual_seed(0)  # issue kornia#2027
```
Comment (johnnv1) [inline test/tracking/test_planar_tracking.py:45]: continue to fail on macos https://github.com/kornia/kornia/actions/runs/3639450296/jobs/6142845358#step:4:9583

**Comment by johnnv1** [comment]
seems to work 1x, I will rerun the CI some times before merge

**Review by edgarriba — APPROVED** [review]

**Comment by edgarriba** [comment]
@johnnv1 amazing!! All green lights 🚀🚀
